### ISSUE TITLE
URL Check

### ISSUE THREAD

**Issue by dulanshuangqiao** [openssl/openssl#25739]
I used OpenSSL 3.0.2 15 Mar 2022 (Library: OpenSSL 3.0.2 15 Mar 2022)
Under this version, the test case conversion was successfully completed by executing the certificate format conversion command openssl x509 -inform DER -in fd.der -outform PEM -out fd.pem
As a comparison, Golang's crypto/x509 library threw an error for this test case。Error message：
parse "https://1kYj\\[@.cfZGv3T_Tr.D?/zrm3/4WA/Ir}BQ/yR]/0[g?<tX=uR?&K'O={d2}&sG?rLi=<}e>": net/url: invalid userinfo
This test case contains a malformed URL as the SAN extension value. 
Golang verifies that the URL is formatted correctly and throws an error. It now appears that openssl x509 does not provide relevant checks. 
At the same time, the openssl x509 -noout -text -in filename command also successfully parses the extension content and displays the malformed URL
[Test_Cases.zip](https://github.com/user-attachments/files/17442238/Test_Cases.zip)
![Golang](https://github.com/user-attachments/assets/f092ffc6-959d-46ed-936a-b549ed0824f6)
![openssl](https://github.com/user-attachments/assets/74ed58b5-4387-4696-9552-cd1b90f29bf9)

**Comment by mattcaswell** [comment]
> It now appears that openssl x509 does not provide relevant checks.

I agree it doesn't appear that OpenSSL performs any checks on the URI format. Looking at the code URIs are encoded as IA5Strings, so as long as it is one of those then we are happy.

The RFC says this:

>    When the subjectAltName extension contains a URI, the name MUST be
>    stored in the uniformResourceIdentifier (an IA5String).  The name
>    MUST NOT be a relative URI, and it MUST follow the URI syntax and
>    encoding rules specified in [[RFC3986](https://datatracker.ietf.org/doc/html/rfc3986)].  The name MUST include both a
>    scheme (e.g., "http" or "ftp") and a scheme-specific-part.  URIs that
>    include an authority ([[RFC3986], Section 3.2](https://datatracker.ietf.org/doc/html/rfc3986#section-3.2)) MUST include a fully
>    qualified domain name or IP address as the host.  Rules for encoding
>    Internationalized Resource Identifiers (IRIs) are specified in
>    [Section 7.4](https://datatracker.ietf.org/doc/html/rfc5280#section-7.4)

I would classify this as a missing feature

**Comment by 20ioppolan** [comment]
I am working on implementing the check

**Comment by t8m** [comment]
Please note we are looking for community help implementing this issue. Please comment if you are interested in implementing this feature or if you have reasons why this feature is important enough so we should consider implementing it with higher priority than other requested features. The issue will be closed after 2 months otherwise.

**Comment by t8m** [comment]
This issue was closed because it was marked with the help wanted label and there was no interest in implementing this issue or any other activity leading to implementing or re-prioritizing it for more than 2 months.
